### ISSUE TITLE
[🐛 Bug Report]: ImportError: cannot import name 'Aggregation' from 'neuralogic.core.settings'

### ISSUE THREAD

**Issue by albertoperdomo2** [LukasZahradnik/PyNeuraLogic#30]
### Describe the bug

When trying to build the model from a template, using the PyG backend, it throws the following error:
```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    model = template.build(Backend.PYG, settings)
  File "/usr/local/lib/python3.8/site-packages/neuralogic/core/template.py", line 124, in build
    return get_neuralogic_layer(backend)(self.module_list)
  File "/usr/local/lib/python3.8/site-packages/neuralogic/nn/__init__.py", line 19, in get_neuralogic_layer
    from neuralogic.nn.native.pyg import NeuraLogic
  File "/usr/local/lib/python3.8/site-packages/neuralogic/nn/native/pyg.py", line 4, in <module>
    from neuralogic.core.settings import Activation, Aggregation
ImportError: cannot import name 'Aggregation' from 'neuralogic.core.settings' (/usr/local/lib/python3.8/site-packages/neuralogic/core/settings/__init__.py)
```

### Steps to reproduce the behavior

This is the script that is being used:
```
from neuralogic.core import Relation, Dataset
from neuralogic.core import Template
from neuralogic.utils.templates import GCNConv, TemplateList
from neuralogic.core import Backend
from neuralogic.core import Settings, Optimizer

dataset = Dataset()
dataset.add_example([
    Relation.edge(1, 2), Relation.edge(2, 1), Relation.edge(1, 3),
    Relation.edge(3, 1), Relation.edge(2, 3), Relation.edge(3, 2),

    Relation.feature(1)[0],
    Relation.feature(2)[1],
    Relation.feature(3)[-1],
])

template = Template(module_list=TemplateList([
    GCNConv(in_channels=5, out_channels=5),
    GCNConv(in_channels=5, out_channels=1),
]))

settings = Settings(learning_rate=0.01, optimizer=Optimizer.SGD, epochs=100)

model = template.build(Backend.PYG, settings)
```

### Expected behavior

The expected behaviour would be the creation of the model from the specified template. 

### Environment

macOS Monterey Version 12.2.1 (21D62)
python 3.8.12
neuralogic 0.1.3
torch-geometric 2.0.3

**Comment by LukasZahradnik** [comment]
Hello, thank you for the bug report and for trying out PyNeuraLogic! 

I released a new version (0.1.4) with the bug being fixed together with some other features/bug fixes.

Thanks to your issue I noticed a few discrepancies in [our documentation](https://pyneuralogic.readthedocs.io/en/latest/quick_start.html#logic-representation) (fixed), namely the dimension of `in_channels` of the first `GCNConv` module (it should be `in_channels=1`). You will also have to rename all `Relation.feature` to `Relation.node_feature` or specify `feature_name` as `"feature"` while instantiating `TemplateList` (it was outdated in the documentation).


I also noticed in your code that you are using a dataset in relational form for the PyG backend. PyNeuraLogic only supports using PyG backend with a [tensor representation](https://pyneuralogic.readthedocs.io/en/latest/quick_start.html#tensor-representation) (`features` and `edge_index` being torch tensor and not python list).

I would recommend you to switch to the Java backend, as it is the primary and the most polished backend of PyNeuraLogic. 
This way, you can fully utilize all possibilities of PyNeuraLogic.

If you decide to go with a different backend (other than PyG), don't forget to add some queries to the dataset; otherwise, PyNeuraLogic will not know labels and outputs. We omitted queries in the documentation for clarity.

You can also check out some of our [example jupyter notebooks](https://pyneuralogic.readthedocs.io/en/latest/examples.html) that might help with using PyNeuraLogic. This framework is quite different from other tensor-based frameworks, so it might take a while to get used to it.

Thanks again for the report. If you need any help or have some other issues, do not hesitate to open a topic in discussions or an issue.

**Comment by albertoperdomo2** [comment]
Thanks for the fast reply and the corrections to my example. I have no problem with using Java backend, so I'll proceed as you mentioned.
